Sources/_SubprocessCShims/process_shims.c

```diff
@@ -664,6 +504,10 @@ int _subprocess_fork_exec(
         // If we reached this point, something went wrong
         write_error_and_exit;
     } else {
+        int _pidfd = _pidfd_open(childPid);
```

Comment:
Could we use `clone` + `CLONE_PIDFD` (Linux 5.2) instead of `fork` + `pidfd_open`? Like FreeBSD's `pdfork`, this avoids races since combining the latter two functions is not atomic.

Reply:
Good call. Unfortunately it still needs to be conditionalized due to kernel version requirements.